src/Microsoft.Owin.Security.Authorization.Mvc/ResourceAuthorizeAttribute.cs

```diff
@@ -45,11 +45,12 @@ protected override bool AuthorizeCore(HttpContextBase httpContext)
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            var controller = httpContext.Items[s_controllerKey] as IAuthorizationController;
+            var filterContext = (System.Web.Mvc.AuthorizationContext)httpContext.Items[s_authorizationContextKey];
+            var controller = filterContext.Controller as IAuthorizationController;
```

Comment:
The filterContext could be null here.
```suggestion
            var controller = filterContext?.Controller as IAuthorizationController;
```